pandora-server-directory/src/account/account.ts

```diff
@@ -115,11 +115,34 @@ export class Account {
 			settings.visibleRoles = uniq(settings.visibleRoles.filter((role) => this.roles.isAuthorized(role)));
 		}
 
+		const db = GetDatabase();
+		const update: Parameters<(typeof db)['updateAccountData']>['1'] = {};
+
+		const now = Date.now();
+		for (const [key, value] of KnownObject.entries(this.data.settingsCooldowns)) {
```

Comment:
This should iterate over `ACCOUNT_SETTINGS_LIMITED_LIMITS` keys instead, because the `data.settingsCooldowns` is partial - it could skip keys if they were never set.
The main bit is that `settingsCooldowns` write needs to happen, which it currently doesn't for a new account.

Maybe it would be simpler iterating over the new `settings` object and checking if the key is a limited setting?